### DISCUSSION TITLE
Module not found: Can't resolve 'fs'

### DISCUSSION THREAD

**Discussion by kliu-tkww** [vercel/next.js#23850] [Help]
### What version of Next.js are you using?

v9.5.5

### What version of Node.js are you using?

v12.18.0

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

yarn start

### Describe the Bug

```
error - ./node_modules/@xo-union/tk-analytics/lib/createSegmentSnippet/index.cjs.js:33:0
Module not found: Can't resolve 'fs'

```
 use the following code
```
module.exports = {
  webpack: (config, { isServer }) => {
    // Fixes npm packages that depend on `fs` module
    if (!isServer) {
      config.node = {
        fs: 'empty'
      }
    }

    return config
  }
}
```
The following error will occur
```
warn  - ./node_modules/uglify-js/tools/node.js
Critical dependency: the request of a dependency is an expression
```

### Expected Behavior

normal operation

### To Reproduce

I'm sorry, because it's an internal project, I can't provide the code

**Comment by codinsonn** [comment]
Any solution yet? (I'm using a third party library in API route that apparently has need of `fs`)

**Comment by MoOx** [comment]
You are probably not following the ["rule" of writing Node code in `pages` folder only](https://github.com/vercel/next.js/issues/16153#issuecomment-673374488).
